persist/storer.go

```diff
@@ -238,6 +244,34 @@ func (j *fileStore) loadFromDisk() error {
 	return nil
 }
 
+func (j *fileStore) deleteOldEntries(ttl time.Duration) error {
+	j.locker.Lock()
+	defer j.locker.Unlock()
+
+	entry := entryTimestamp{}
+
+	unixTTL := int64(ttl.Seconds())
+	unixNow := time.Now().Unix()
+
+	var expiredKeys []string
+
+	for key, val := range j.Data {
+		if err := json.Unmarshal(val, &entry); err != nil {
+			return err
+		}
+
+		if (unixNow - entry.Timestamp) > unixTTL {
+			expiredKeys = append(expiredKeys, key)
+		}
+	}
+
+	for _, key := range expiredKeys {
+		delete(j.Data, key)
+	}
```

Comment:
Why do we store the keys to be deleted and then iterate over them, instead of deleting them in-place?

Reply:
Oh i thought it was not safe but it actually is safe to do it 
https://stackoverflow.com/questions/23229975/is-it-safe-to-remove-selected-keys-from-map-within-a-range-loop